### ISSUE TITLE
Validation loss not computed

### ISSUE THREAD

**Issue by nitinnairk** [lopuhin/transformer-lm#16]
Is there a reason why validation loss is not computed nor logged when the model is trained with more than one GPU?

**Comment by lopuhin** [comment]
IIRC I didn't manage to make it work for some reason, so I think I ended up running validation from a separate process - but also I didn't get to train long enough to overfit.

**Comment by nitinnairk** [comment]
Could you share that validation script?
I'm using this GPT model to train a different language altogether. Therefore, having the validation loss would be of great help!

**Comment by lopuhin** [comment]
If you pass ``--only-validate`` option, then the validation loss would be computed - the only caveat is that you need to make sure you're not using multiple GPUs (e.g. limit to one gpu with ``CUDA_VISIBLE_DEVICES=0`` environment variable)::

https://github.com/lopuhin/transformer-lm/blob/fa3f529ff300a30cd984ea72a4e23b525b6e3f52/lm/main.py#L251-L256

**Comment by nitinnairk** [comment]
Got it! Thanks
Should I close this issue given the actual issue of multi-GPU validation computation is still not solved?

**Comment by lopuhin** [comment]
Let's leave it open until it's supported. Thanks for report, I hope this issue will be useful in the meantime.
